web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -156,77 +213,86 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
-                        />
-                      </div>
-                      <Title variant="h6" sx={{ mr: 4 }}>
-                        =
-                      </Title>
-                      <div
-                        className={cx(styles.flexColumn, styles.marginRight)}
+                      <Body
+                        size="md"
+                        sx={{ color: 'primary.light', fontWeight: 700, mb: 3 }}
                       >
-                        <Box
-                          sx={{
-                            display: 'flex',
-                            alignItems: 'baseline',
-                            flexWrap: 'wrap',
-                          }}
+                        Batch denom:{' '}
+                        <Body
+                          sx={{ fontWeight: 'normal', display: 'inline-block' }}
+                        >
+                          {selectedSellOrder?.batchDenom}
+                        </Body>
+                      </Body>
+                      <div className={styles.creditWidget}>
+                        <div className={styles.marginRight}>
+                          <Field
+                            className={cx(styles.creditInput)}
+                            component={NumberTextField}
+                            name="creditCount"
+                            min={1}
+                            max={selectedSellOrder?.quantity}
+                          />
+                        </div>
+                        <Title variant="h6" sx={{ mr: 4 }}>
+                          =
+                        </Title>
+                        <div
+                          className={cx(styles.flexColumn, styles.marginRight)}
                         >
-                          <Box sx={{ display: 'flex', alignItems: 'baseline' }}>
-                            <RegenTokenIcon className={styles.regenIcon} />
-                            <Title variant="h4" sx={{ mr: 1.5 }}>
-                              {values.creditCount * (initialValues?.price ?? 0)}
-                            </Title>
+                          <Box
+                            sx={{
+                              display: 'flex',
+                              alignItems: 'baseline',
+                              flexWrap: 'wrap',
+                            }}
+                          >
+                            <Box
+                              sx={{ display: 'flex', alignItems: 'baseline' }}
+                            >
+                              <RegenTokenIcon className={styles.regenIcon} />
+                              <Title variant="h4" sx={{ mr: 1.5 }}>
+                                {values.creditCount *
+                                  ((selectedSellOrder?.askAmount &&
+                                    microToDenom(
+                                      selectedSellOrder.askAmount,
+                                    )) ||
+                                    0) || '-'}
```

Comment:
Should the undefined case be handled by `microToDenom` ?
It would simplify the usage:

```suggestion
                                {values.creditCount *
                                    microToDenom(
                                      selectedSellOrder.askAmount,
                                    ) || '-'}
```

Reply:
handling undefined now.